### DISCUSSION TITLE
Fancy Zones like pre-configured layout similar to XMonad

### DISCUSSION THREAD

**Discussion by MultiCoreNop** [i3/i3#5437] [Q&A] [answered]
I read through the docs, but could not figure out whether this is possible with i3wm:

Imagine a huge monitor (e.g. a whole wall). If you open your first application, you probably wouldn't want that application to go "fullscreen". Is there a way to preconfigure a layout (maybe XMonad-style) like "1st window goes here, 2nd..5th vertically right to the first, ..."?
Another way to put it would be "Microsoft's Fancy Zones with a predefined order to fill those zones when new windows get opened".

**Comment by Airblader** [comment, marked as answer]
This isn't possible with i3.

**Comment by vincentbernat** [comment]
It may not fit exactly what you want, but depending if you know in advance which applications are going to take space, you can save an existing layout and restore it. Check https://i3wm.org/docs/layout-saving.html.

**Comment by neg-serg** [comment]
There is no any direct way to do this.
You need a quiet lot of sophisticated i3-ipc magic to do this and result may always be not perfect. But maybe I will try to implement this later via i3ipc-python module.
